### PR TITLE
Local support 

### PR DESCRIPTION
**Pull request by naveed-ahmad**
Added internationalizing 💯  #245 

### Added support for:  
* Urdu 
* Arabic  
* English

And opened the door for others languages :)

### CODE REVIEW

**Comment by ahmedre** [comment]
Deployed to: http://staging.quran.com:32993

**Comment by mmahalwy** [comment]
this is so awesome! So excited for this :)

**Comment by mmahalwy** [comment]
I can probably feed the English into Google Translate and then we will have an arabic locale too!

**Comment by ahmedre** [comment]
Deployed to: http://staging.quran.com:33005

**Comment by ahmedre** [comment]
Deployed to: http://staging.quran.com:33006

**Comment by ahmedre** [comment]
Deployed to: http://staging.quran.com:33007

**Comment by mmahalwy** [comment]
Tests are very unhappy with you right now @naveed-ahmad

**Comment by ahmedre** [comment]
Deployed to: http://staging.quran.com:33008

**Comment by ahmedre** [comment]
Deployed to: http://staging.quran.com:33009

**Comment by ahmedre** [comment]
Deployed to: http://staging.quran.com:33014

**Comment by ahmedre** [comment]
Deployed to: http://staging.quran.com:33015

**Comment by ahmedre** [comment]
Deployed to: http://staging.quran.com:33016

**Comment by ahmedre** [comment]
Deployed to: http://staging.quran.com:33017

**Comment by ahmedre** [comment]
Deployed to: http://staging.quran.com:33020

**Comment by sabeurthabti** [comment]
@naveed-ahmad 
``` bash 
Chrome 55.0.2883 (Mac OS X 10.12.1): Executed 3 of 8 SUCCESS (0 secs / 0.019 secs)
20 12 2016 00:01:16.454:INFO [karma]: Delaying execution, these browsers are not ready: Chrome 55.0.2883 (Mac OS X 10.Chrome 55.0.2883 (Mac OS X 10.12.1) <ContentDropdown /> should remove all content FAILED
	Invariant Violation: [React Intl] Could not find required `intl` object. <IntlProvider> needs to exist in the component ancestry.
	    at invariant (webpack:///~/invariant/browser.js:40:0 <- tests.webpack.js:5463:15)
	    at invariantIntlContext (webpack:///~/react-intl/lib/index.es.js:606:0 <- tests.webpack.js:11585:54)
	    at new InjectIntl (webpack:///~/react-intl/lib/index.es.js:679:0 <- tests.webpack.js:11658:13)
	    at ReactCompositeComponentWrapper.mountComponent (webpack:///~/react/lib/ReactCompositeComponent.js:148:0 <- tests.webpack.js:39146:18)
	    at ReactCompositeComponentWrapper.wrapper [as mountComponent] (webpack:///~/react/lib/ReactPerf.js:66:0 <- tests.webpack.js:2892:21)
	    at Object.mountComponent (webpack:///~/react/lib/ReactReconciler.js:37:0 <- tests.webpack.js:6236:35)
	    at ReactDOMComponent.mountChildren (webpack:///~/react/lib/ReactMultiChild.js:241:0 <- tests.webpack.js:80622:44)
	    at ReactDOMComponent._createContentMarkup (webpack:///~/react/lib/ReactDOMComponent.js:591:0 <- tests.webpack.js:78389:32)
	    at ReactDOMComponent.mountComponent (webpack:///~/react/lib/ReactDOMComponent.js:479:0 <- tests.webpack.js:78277:29)
	    at Object.mountComponent (webpack:///~/react/lib/ReactReconciler.js:37:0 <- tests.webpack.js:6236:35)
Chrome 55.0.2883 (Mac OS X 10.12.1): Executed 8 of 8 (1 FAILED) (3.858 secs / 0.033 secs)
Chrome 55.0.2883 (Mac OS X 10.12.1) <ContentDropdown /> should remove all content FAILED
	Invariant Violation: [React Intl] Could not find required `intl` object. <IntlProvider> needs to exist in the component ancestry.
	    at invariant (webpack:///~/invariant/browser.js:40:0 <- tests.webpack.js:5463:15)
	    at invariantIntlContext (webpack:///~/react-intl/lib/index.es.js:606:0 <- tests.webpack.js:11585:54)
	    at new InjectIntl (webpack:///~/react-intl/lib/index.es.js:679:0 <- tests.webpack.js:11658:13)
	    at ReactCompositeComponentWrapper.mountComponent (webpack:///~/react/lib/ReactCompositeComponent.js:148:0 <- tests.webpack.js:39146:18)
	    at ReactCompositeComponentWrapper.wrapper [as mountComponent] (webpack:///~/react/lib/ReactPerf.js:66:0 <- tests.webpack.js:2892:21)
	    at Object.mountComponent (webpack:///~/react/lib/ReactReconciler.js:37:0 <- tests.webpack.js:6236:35)
	    at ReactDOMComponent.mountChildren (webpack:///~/react/lib/ReactMultiChild.js:241:0 <- tests.webpack.js:80622:44)
	    at ReactDOMComponent._createContentMarkup (webpack:///~/react/lib/ReactDOMComponent.js:591:0 <- tests.webpack.js:78389:32)
	    at ReactDOMComponent.mountComponent (webpack:///~/react/lib/ReactDOMComponent.js:479:0 <- tests.webpack.js:78277:29)
	    at Object.mountComponent (webpack:///~/react/lib/ReactReconciler.js:37:0 <- tests.webpack.js:6236:35)
Chrome 55.0.2883 (Mac OS X 10.12.1): Executed 8 of 8 (1 FAILED) (1.27 secs / 0.039 secs)
```

Within PageBreak/spec.js, I am unable to call enzyme directly. PageBreak doesn't use React-Intel, can we just use Enzme without your helper to test this?

**Comment by ahmedre** [comment]
Deployed to: http://staging.quran.com:33021

**Comment by mmahalwy** [comment]
this is just missing reviews right @naveed-ahmad ? No longer WIP?

**Comment by ahmedre** [comment]
Deployed to: http://staging.quran.com:33026

**Comment by ahmedre** [comment]
Deployed to: http://staging.quran.com:33027

**Comment by naveed-ahmad** [comment]
@mmahalwy @sabeurthabti ready for review

**Comment by naveed-ahmad** [comment]
Thanks for reviewing :) @arshdkhn1 maybe you can add hindi local

File: src/components/Ayah/index.js [hunk 1]
```diff
@@ -204,15 +208,25 @@ export default class Ayah extends Component {
   }
 
   renderPlayLink() {
-    const { isSearched, ayah } = this.props;
+    const { isSearched, ayah, currentAyah, isPlaying } = this.props;
+    const playing = ayah.ayahKey == currentAyah && isPlaying;
+    let icon = <i className="ss-icon ss-play" />
```
Comment (naveed-ahmad) [inline src/components/Ayah/index.js]: ✂️

File: src/components/Ayah/index.js [hunk 2]
```diff
@@ -233,17 +247,17 @@ export default class Ayah extends Component {
...
+    if (isSearched || !isAuthenticated) return false;
 
     if (bookmarked) {
       return (
         <a
           onClick={() => bookmarkActions.removeBookmark(ayah.ayahKey)}
           className="text-muted"
         >
-          <strong><i className="ss-icon ss-bookmark" /> Bookmarked</strong>
+          <strong><i className="ss-icon ss-bookmark" /> Bookmarked </strong>
```
Comment (naveed-ahmad) [inline src/components/Ayah/index.js]: localized this as well

File: tests/helpers/intl-enzyme-test-helper.js [hunk 3]
```diff
@@ -0,0 +1,39 @@
...
+var intl  = intlProvider.getChildContext();
+
+/**
+ * When using React-Intl `injectIntl` on components, props.intl is required.
+ */
+function nodeWithIntlProp(node) {
+  return React.cloneElement(node,  intl );
+}
+
+/**a
```
Comment (naveed-ahmad) [inline tests/helpers/intl-enzyme-test-helper.js]: ✂️ a

**Comment by ahmedre** [comment]
Deployed to: http://staging.quran.com:33028

File: src/components/Footer/index.js [hunk 4]
```diff
@@ -13,19 +16,42 @@ const Footer = () => (
       <Col md={10} mdOffset={1}>
         <Row>
           <Col md={2} sm={4} xs={12} className={styles.about}>
-            <p>Navigate</p>
+            <p className={styles.header}>Navigate</p>
```
Comment (mmahalwy) [inline src/components/Footer/index.js]: Requires translation please :)

File: src/components/Footer/index.js [hunk 5]
```diff
@@ -61,13 +93,27 @@ const Footer = () => (
...
+              <LocaleSwitcher/>
+
+              <p className="monserrat">
+                <LocaleFormattedMessage
+                  id={'nav.aboutQuranProject'}
+                defaultMessage={'Quran.com (also known as The Noble Quran, Al Quran, Holy Quran, Koran) is a pro bono project.'}
+                />
+              </p>
+
+              <p className="monserrat">&copy; QURAN.COM. ALL RIGHTS RESERVED 2016</p>
```
Comment (mmahalwy) [inline src/components/Footer/index.js]: should this be translated?

**Comment by mmahalwy** [comment]
Thank you @naveed-ahmad !

File: src/components/Footer/index.js (hunk 5 above)
Comment (naveed-ahmad) [inline src/components/Footer/index.js]: Yep, we need to translate this and other static pages. Next PR, it was already became huge.

File: src/config.js [hunk 6]
```diff
@@ -90,6 +90,7 @@ module.exports = Object.assign({
       /* SEO: https://developers.google.com/structured-data/slsb-overview#markup_examples */
       /* SEO: https://developers.google.com/structured-data/site-name#markup_requirements */
       script: [
+        { src: "https://cdn.polyfill.io/v2/polyfill.min.js?features=Intl.~locale.en"},
```
Comment (naveed-ahmad) [inline src/config.js]: react-intl depends on built-in `intl` api, which is missing in old browser and safari. This polyfill add missing `intl` for those browsers.  

https://github.com/yahoo/react-intl/wiki#i18n-in-javascript

File: src/components/Audioplayer/RepeatDropdown/index.js [hunk 7]
```diff
@@ -107,7 +119,10 @@ export default class RepeatButton extends Component {
 
     return (
       <Col md={12} style={{paddingTop: 15}}>
-        Ayah: <br />
+        <LocaleFormattedMessage
+          id={ "player.currentAyah" }
```
Comment (naveed-ahmad) [inline src/components/Audioplayer/RepeatDropdown/index.js]: No, Ill love to learn and follow best practices. So we don't need {} for string literals

**Pull request merged**